framework/contrib/ice_updater/include/ErrorLogger.h

```diff
@@ -0,0 +1,84 @@
+/*******************************************************************************
+ * Copyright (c) 2015, UT-Battelle, LLC.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *  notice, this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *  notice, this list of conditions and the following disclaimer in
+ *  the documentation and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its
+ *  contributors may be used to endorse or promote products derived
+ *  from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+ * OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+*******************************************************************************/
+
+#ifndef ERRORLOGGER_H
+#define ERRORLOGGER_H
+
+#include <vector>
+#include <string>
+
+using namespace std;
+
+/**
+ * The ErrorLogger class has functions to store errors in a Vector&lt;string&gt;
+ * attribute and then dump those errors to a file.
+ */
+class ErrorLogger
+{
+
+private:
+
+  /**
+   * A ErrorVector object to contain string errors.
+   */
+  vector<string> errorVector;
+
+public:
+
+  /**
+   * The Constructor.
+   */
+  ErrorLogger();
+
+  /**
+   * The Destructor.
+   */
+  ~ErrorLogger();
+
+  /**
+   * Appends an error string to the errorVector attribute.
+   *
+   * @param error An error string.
+   */
+  void logError(string error);
+
+  /**
+   * Dumps the errors in errorVector into a file with a filename format
+   * "updatererrors_&lt;timestamp&gt;.log".
+   *
+   * @return The name of the error log file.
+   */
+  string dumpErrors();
+
```

Comment:
You have a ton of empty lines after ```{``` and before ```}``` which is a bit at odds with our [code standards(http://mooseframework.org/wiki/CodeStandards/). Same goes for single line doc-comments not using ```///```, using ```using std;```, not having a whitespace after ```//```, and indentation (switch/case for example).
But I know I _am_ being picky here ;-)